codice/ddf
# DDF-04824 Adds Storybook to DDF

#### What does this PR do?

- Bump version of ace to bring in new storybook command
- Bump version of node/yarn for storybook compatibility
- Bump node-sass version to be compatible with the new node version
- Adds prop-types to allow better documentation for js react components
- Updates components to be more modular
- Adds a few stories

[demo](https://djblue.github.io/storybook-example/)

#### Who is reviewing it? 

@adimka 
@andrewkfiedler 
@samuelechu 
@gordocanchola 
@pklinef 

#### Select relevant component teams: 

@codice/ui 

#### How should this be tested?

- `mvn clean install` ui directory
- yarn start 
  - use location component in query tool and ensure it still works
- yarn storybook
  - ensure components work in isolation

#### What are the relevant tickets?

Fixes: #4824

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
